src/main/java/ru/sbt/mipt/oop/eventcircle/EventCircleImpl.java

```diff
@@ -0,0 +1,35 @@
+package ru.sbt.mipt.oop.eventcircle;
+
+import ru.sbt.mipt.oop.Event;
+import ru.sbt.mipt.oop.SmartHome;
+import ru.sbt.mipt.oop.eventhandler.AllLightEventHandler;
+import ru.sbt.mipt.oop.eventhandler.DoorEventHandler;
+import ru.sbt.mipt.oop.eventgenerator.EventGenerator;
+import ru.sbt.mipt.oop.eventhandler.EventHandler;
+import ru.sbt.mipt.oop.eventhandler.LightEventHandler;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class EventCircleImpl implements EventCircle{
+	private final SmartHome smartHome;
+	private final EventGenerator eventGenerator;
+	private final List<EventHandler> eventHandlers;
+
+	public EventCircleImpl(SmartHome smartHome, EventGenerator eventGenerator) {
+		this.smartHome = smartHome;
+		this.eventGenerator = eventGenerator;
+		eventHandlers = Arrays.asList(new DoorEventHandler(smartHome), new LightEventHandler(smartHome), new AllLightEventHandler(smartHome));
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D0%BF%D0%B5%D1%80%D0%B5%D0%B4%D0%B0%D0%B2%D0%B0%D0%B9%D1%82%D0%B5-%D0%B7%D0%B0%D0%B2%D0%B8%D1%81%D0%B8%D0%BC%D0%BE%D1%81%D1%82%D0%B8-%D1%81%D0%BD%D0%B0%D1%80%D1%83%D0%B6%D0%B8